apps/meteor/app/livechat/server/lib/contacts/getContacts.ts

```diff
@@ -0,0 +1,49 @@
+import type { IUser } from '@rocket.chat/core-typings';
+import { LivechatContacts, Users } from '@rocket.chat/models';
+import type { PaginatedResult, ILivechatContactWithManagerData } from '@rocket.chat/rest-typings';
+import type { Sort } from 'mongodb';
+
+export type GetContactsParams = {
+	searchText?: string;
+	count: number;
+	offset: number;
+	sort: Sort;
+	unknown?: boolean;
+};
+
+export async function getContacts(params: GetContactsParams): Promise<PaginatedResult<{ contacts: ILivechatContactWithManagerData[] }>> {
+	const { searchText, count, offset, sort, unknown } = params;
+
+	const { cursor, totalCount } = LivechatContacts.findPaginatedContacts(
+		{ searchText, unknown },
+		{
+			limit: count,
+			skip: offset,
+			sort: sort ?? { name: 1 },
+		},
+	);
+
+	const [rawContacts, total] = await Promise.all([cursor.toArray(), totalCount]);
+
+	const managerIds = [...new Set(rawContacts.map(({ contactManager }) => contactManager))];
+	const managersData = await Users.findByIds<Pick<IUser, '_id' | 'name' | 'username'>>(managerIds, {
+		projection: { name: 1, username: 1 },
+	}).toArray();
+	const mappedManagers = Object.fromEntries(managersData.map((manager) => [manager._id, manager]));
```

Comment:
You can do this map before calling the toArray()

Reply:
I don't know if I see any benefit on this one 🤔

Reply:
I was trying to look for a link to paste here. I remember this was "nicer" on memory because you don't have to do the map again after doing the toArray on large results, doing the map before the toArray helps you to get the data already mapped when the array is created.

But cannot find anything regarding that, maybe @MarcosSpessatto do you remember something? Or i'm crazy? lel

Reply:
hmmm got it. I've just applied this change

Reply:
In this specific case doing the map after the toArray was actually slightly better (since this map increases the total size of each item instead of reducing it), but the difference between the two was negligible (just one extra string per item).